packages/liveblocks-core/src/http-client.ts

```diff
@@ -1,11 +1,1400 @@
+        id: string;
+      }[];
+      createdAt: Date;
+    }[];
+    requestedAt: Date;
+  }>;
+
+  listTextVersionsSince({
+    roomId,
+    since,
+    signal,
+  }: {
+    roomId: string;
+    since: Date;
+    signal?: AbortSignal;
+  }): Promise<{
+    versions: {
+      type: "historyVersion";
+      kind: "yjs";
+      id: string;
+      authors: {
+        id: string;
+      }[];
+      createdAt: Date;
+    }[];
+    requestedAt: Date;
+  }>;
+
+  streamStorage(options: {
+    roomId: string;
+    getAuthValue: () => Promise<AuthValue>;
+  }): Promise<IdTuple<SerializedCrdt>[]>;
+
+  sendMessages<P extends JsonObject, E extends Json>(options: {
+    roomId: string;
+    getAuthValue: () => Promise<AuthValue>;
+    nonce: string | undefined;
+    messages: ClientMsg<P, E>[];
+  }): Promise<Response>;
+}
+
+export interface NotificationHttpApi {
+  getInboxNotifications<M extends BaseMetadata>(options?: {
+    cursor?: string;
+  }): Promise<{
+    inboxNotifications: InboxNotificationData[];
+    threads: ThreadData<M>[];
+    nextCursor: string | null;
+    requestedAt: Date;
+  }>;
+
+  getInboxNotificationsSince<M extends BaseMetadata>(options: {
+    since: Date;
+    signal?: AbortSignal;
+  }): Promise<{
+    inboxNotifications: {
+      updated: InboxNotificationData[];
+      deleted: InboxNotificationDeleteInfo[];
+    };
+    threads: {
+      updated: ThreadData<M>[];
+      deleted: ThreadDeleteInfo[];
+    };
+    requestedAt: Date;
+  }>;
+
+  getUnreadInboxNotificationsCount(): Promise<number>;
+
+  markAllInboxNotificationsAsRead(): Promise<void>;
+
+  markInboxNotificationAsRead(inboxNotificationId: string): Promise<void>;
+
+  deleteAllInboxNotifications(): Promise<void>;
+
+  deleteInboxNotification(inboxNotificationId: string): Promise<void>;
+}
+
+export interface ClientHttpApi extends RoomHttpApi, NotificationHttpApi {
+  getUserThreads_experimental<M extends BaseMetadata>(options?: {
+    cursor?: string;
+    query?: {
+      resolved?: boolean;
+      metadata?: Partial<QueryMetadata<M>>;
+    };
+  }): Promise<{
+    threads: ThreadData<M>[];
+    inboxNotifications: InboxNotificationData[];
+    nextCursor: string | null;
+    requestedAt: Date;
+    permissionHints: Record<string, Permission[]>;
+  }>;
+
+  getUserThreadsSince_experimental<M extends BaseMetadata>(options: {
+    since: Date;
+    signal?: AbortSignal;
+  }): Promise<{
+    inboxNotifications: {
+      updated: InboxNotificationData[];
+      deleted: InboxNotificationDeleteInfo[];
+    };
+    threads: {
+      updated: ThreadData<M>[];
+      deleted: ThreadDeleteInfo[];
+    };
+    requestedAt: Date;
+    permissionHints: Record<string, Permission[]>;
+  }>;
+}
+
+export function createHttpClient({
+  baseUrl,
+  authManager,
+  fetchPolyfill,
+}: {
+  baseUrl: string;
+  authManager: AuthManager;
+  fetchPolyfill: typeof fetch;
+}): ClientHttpApi {
+  async function getAuthValueForRoom(roomId: string) {
+    // TODO: Use the right scope
```

Comment:
I've seen this comment before, so I know you copied it. Do you know what was meant by this?

Reply:
Unfortunately, I do not. Guillaume might be the best person to ask about this. I asked Marc too and we both are unsure what the correct scope would be here, if it is not already.